src/Cake.Common/Tools/Chocolatey/Pack/ChocolateyPacker.cs

```diff
@@ -0,0 +1,185 @@
+            {
+                throw new CakeException("Required setting Files not specified.");
+            }
+
+            Pack(settings, () => _processor.Process(settings));
+        }
+
+        /// <summary>
+        /// Creates a Chocolatey package from the specified Nuspec file.
+        /// </summary>
+        /// <param name="nuspecFilePath">The nuspec file path.</param>
+        /// <param name="settings">The settings.</param>
+        public void Pack(FilePath nuspecFilePath, ChocolateyPackSettings settings)
+        {
+            if (nuspecFilePath == null)
+            {
+                throw new ArgumentNullException("nuspecFilePath");
+            }
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+
+            Pack(settings, () => _processor.Process(nuspecFilePath, settings));
+        }
+
+        private void Pack(ChocolateyPackSettings settings, Func<FilePath> process)
+        {
+            FilePath procesedNuspecFilePath = null;
+            try
+            {
+                // Transform the nuspec and return the new filename.
+                procesedNuspecFilePath = process();
+
+                // Start the process.
+                Run(settings, GetArguments(procesedNuspecFilePath, settings), settings.ToolPath);
+            }
+            finally
+            {
+                if (procesedNuspecFilePath != null)
+                {
+                    // Delete the processed file.
+                    var file = _fileSystem.GetFile(procesedNuspecFilePath);
+                    if (file.Exists)
+                    {
+                        file.Delete();
+                    }
+                }
+            }
+        }
+
+        private ProcessArgumentBuilder GetArguments(FilePath nuspecFilePath, ChocolateyPackSettings settings)
+        {
+            var builder = new ProcessArgumentBuilder();
+            builder.Append("pack");
+
+            // Debug
+            if (settings.Debug)
+            {
+                builder.Append("-d");
+            }
+
+            // Verbose
+            if (settings.Verbose)
+            {
+                builder.Append("-v");
+            }
+
+            // Always say yes, so as to not show interactive prompt
+            builder.Append("-y");
+
+            // Force
+            if (settings.Force)
+            {
+                builder.Append("-f");
+            }
+
+            // Noop
+            if (settings.Noop)
+            {
+                builder.Append("--noop");
+            }
+
+            // Limit Output
+            if (settings.LimitOutput)
+            {
+                builder.Append("-r");
+            }
+
+            // Execution Timeout
+            if (settings.ExecutionTimeout != 0)
+            {
+                builder.Append("--execution-timeout");
+                builder.AppendQuoted(settings.ExecutionTimeout.ToString());
```

Comment:
To avoid any localization issues this should probably be `builder.AppendQuoted(settings.ExecutionTimeout.ToString(CultureInfo.InvariantCulture));`